src/Http/Http/test/StreamPipeWriterTests.cs

```diff
@@ -320,6 +320,24 @@ public void CallCompleteWithoutFlushAndException_DoesNotThrowInvalidOperationExc
             Writer.Complete(new Exception());
         }
 
+        [Fact]
+        public void GetMemorySameAsTheMaxPoolSizeUsesThePool()
+        {
+            var memory = Writer.GetMemory(Pool.MaxBufferSize);
+
+            Assert.Equal(Pool.MaxBufferSize, memory.Length);
+            Assert.Equal(1, Pool.GetRentCount());
+        }
+
+        [Fact]
+        public void GetMemoryBiggerThanPoolSizeAllocatesUnpooledArray()
+        {
+            var memory = Writer.GetMemory(Pool.MaxBufferSize + 1);
```

Comment:
What is the default size of this memory? If it's Int.Max, consider recreating the pool with a different MaxBufferSize.

Reply:
It'll OOM.